docs/src/user_docs/assembly/cryptographic_operations.md

```diff
@@ -90,6 +90,36 @@ Both `hash` and `hmerge` instructions are actually "macro-instructions" which ar
+| eval_circuit <br> - *(1 cycle)*     | [ptr, n_read, n_eval, ...]                                                                        | [ptr, n_read, n_eval, ...]                                                                          | Evaluates an arithmetic circuit, and checks that its output is equal to zero. `ptr` specifies the memory address at which the circuit description is stored with the number of input extension field elements specified by `n_read` and the number of evaluation gates, encoded as base field elements, specified by `n_eval`. |
+| horner_eval_base <br> - *(1 cycle)* | [c7,  c6,  c5,  c4,  c3,  c2,  c1,  c0, - , - , - , - , - , alpha_addr, acc1, acc0, ...]          | [c7,  c6,  c5,  c4,  c3,  c2,  c1,  c0, - , - , - , - , - , alpha_addr, acc1', acc0', ...]          | Performs 8 steps of the Horner evaluation method on a polynomial with coefficients over the base field, i.e., it computes `acc' = (((acc_tmp * alpha + c4) * alpha + c5) * alpha + c6) * alpha + c7` where `acc_tmp := (((acc * alpha + c0) * alpha + c1) * alpha + c2) * alpha + c3`                                          |
+| horner_eval_ext <br> - *(1 cycle)*  | [c3_1, c3_0, c2_1, c2_0, c1_1, c1_0, c0_1, c0_0, - , - , - , - , - , alpha_addr, acc1, acc0, ...] | [c3_1, c3_0, c2_1, c2_0, c1_1, c1_0, c0_1, c0_0, - , - , - , - , - , alpha_addr, acc1', acc0', ...] | Performs 4 steps of the Horner evaluation method on a polynomial with coefficients over the quadratic extension field, i.e., it computes `acc' = (acc_tmp * alpha + c2) * alpha + c3` where `acc_tmp = (acc * alpha + c0) * alpha + c1`                                                                                        |
+
+
+### FRI folding
+
+The following instructions are used during the FRI protocol as part of recursive verification within the Miden VM.
+
+| Instruction                      | Stack_input                                                               | Stack_output                                                                        | Notes                                                                                       |
+| -------------------------------- | ------------------------------------------------------------------------- | ----------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------- |
+| fri_ext2fold4<br>- *(1 cycle)*   | [v7, ..., v0, f_pos, d_seg, poe, e1, e0, a1, a0, layer_ptr, rem_ptr, ...] | [x, x, x, x, x, x, x, x, x, x, layer_ptr + 8, poe^4, f_pos, ne1, ne0, rem_ptr, ...] | Performs one step of FRI folding with folding factor 4 in the quadratic extension field[^1] |
+
+[^1]: In more details:
```

Comment:
Rather than making this a footnote, I'd probably make this just regular text.

Reply:
You mean just under the table (and not inside the table), right?

Reply:
All looks good! Thank you!